Sample-WPF/Controls/FaceDetectionPage.xaml.cs

```diff
@@ -203,69 +204,123 @@ public string SelectedFile
+                                Accessories = $"{GetAccessories(face.FaceAttributes.Accessories)}",
+                                Blur = string.Format("Blur: {0}", face.FaceAttributes.Blur.BlurLevel.ToString()),
+                                Exposure = string.Format("{0}", face.FaceAttributes.Exposure.ExposureLevel.ToString()),
+                                Noise = string.Format("Noise: {0}", face.FaceAttributes.Noise.NoiseLevel.ToString()),
+                                Makeup = string.Format("Makeup: {0}", ((face.FaceAttributes.Makeup.EyeMakeup || face.FaceAttributes.Makeup.LipMakeup) ? "Yes" : "No")),
+                                Hair = string.Format("Hair: {0}", GetHair(face.FaceAttributes.Hair)),
+                            });
+                        }
+
+                        // Convert detection result into UI binding object for rendering
+                        foreach (var face in UIHelper.CalculateFaceRectangleForRendering(faces, MaxImageSize, imageInfo))
+                        {
+                            ResultCollection.Add(face);
+                        }
+                    }
+                    catch (FaceAPIException ex)
+                    {
+                        MainWindow.Log("Response: {0}. {1}", ex.ErrorCode, ex.ErrorMessage);
+                        GC.Collect();
+                        return;
+                    }
+                    GC.Collect();
+                }   
             }
+        }
+
+        private string GetHair(Contract.Hair hair)
+        {
+            if (hair.HairColor.Length == 0)
+            {
+                if (hair.Invisible)
+                    return "Invisible";
+                else
+                    return "Bald";
+            }
+            else
+            {
+                Contract.ColorType returnColor = Contract.ColorType.Unknown;
+                float maxConfidence = 0.0f;
+
+                for (int i = 0; i < hair.HairColor.Length; ++i)
+                {
+                    if (hair.HairColor[i].Confidence > maxConfidence)
+                    {
+                        maxConfidence = (float)hair.HairColor[i].Confidence;
+                        returnColor = hair.HairColor[i].Color;
+                    }
+                }
+
+                return returnColor.ToString();
+            }
+        }
+
+        private string GetAccessories(Contract.Accessory[] accessories)
+        {
+            if (accessories.Length == 0)
+            {
+                return "NoAccessories";
+            }
+
+            string accessoriesType = string.Empty;
+
+            for (int i = 0; i < accessories.Length - 1; ++i)
+            {
+                accessoriesType = accessoriesType + accessories[i].Type.ToString() + " ";
+            }
+
+            accessoriesType = accessoriesType + accessories[accessories.Length - 1].Type.ToString();
+            return "Accessories: "+ accessoriesType;

```

Comment:
I suggest we use `String.Join()` [1] instead of raw concatenation.

[1] https://msdn.microsoft.com/en-us/library/57a79xd0(v=vs.110).aspx